### ISSUE TITLE
alpha in Log space.

### ISSUE THREAD

**Issue by Junyoungpark** [medipixel/rl_algorithms#171]
Hi. I'm just wondering why alpha get updated in 'log' space rather than in its own value.
Was doing that stable for entire training? Or any other reason?

```python
# In https://github.com/medipixel/rl_algorithms/blob/master/algorithms/sac/agent.py

 # train alpha
        if self.hyper_params["AUTO_ENTROPY_TUNING"]:
            alpha_loss = (
                -self.log_alpha * (log_prob + self.target_entropy).detach()
            ).mean()

            self.alpha_optimizer.zero_grad()
            alpha_loss.backward()
            self.alpha_optimizer.step()

            alpha = self.log_alpha.exp()
        else:
            alpha_loss = torch.zeros(1)
            alpha = self.hyper_params["W_ENTROPY"]
```

**Comment by Curt-Park** [comment]
That's because alpha should be greater than or equal to zero due to the dual problem's constraint.
see [(13)](https://arxiv.org/pdf/1812.05905.pdf).

**Comment by Junyoungpark** [comment]
thanks for the comment :)

**Comment by Junyoungpark** [comment]
![derv](https://user-images.githubusercontent.com/3063343/61600843-c818ec00-ac6d-11e9-835c-bff176472e77.png)

Hi, again.

Now I've got a new question on the implementation. The derivative of the alpha in the implementation is a bit different from the original derivation. In the original derivation, the magnitude of d(alpha) is solely depending on the expectation. On the other side, in the implementation, it depends on its current value and expectation. 

It might be good to have an adaptive learning size than fixed (the original one). I'm just wondering that is this also intended?

**Comment by Curt-Park** [comment]
The authors used the adaptive learning size in their [pytorch implementation](https://github.com/vitchyr/rlkit/blob/master/rlkit/torch/sac/sac.py#L99). It tends to show more stable learning as the example below (lunarlander continuous - [full log](https://app.wandb.ai/medipixel_rl/lunarlander_continuous_v2/reports?view=curt-park%2Flog%20alpha%20for%20loss)).

![adaptive_loss_sac](https://user-images.githubusercontent.com/14961526/61617122-08e02780-aca5-11e9-8f97-4e14b6e88583.png)

You can see the non-adaptive learning size causes lots of fluctuations on its alpha loss during learning.

**Comment by Junyoungpark** [comment]
The figure above looks promising for me! 
Thanks a lot

**Comment by Curt-Park** [comment]
You are so welcome.
Is it okay to close this issue?

**Comment by Junyoungpark** [comment]
Sure. Thanks again.
